src/Microsoft.DotNet.XHarness.Common/WasmSymbolicatorBase.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Microsoft.DotNet.XHarness.Common;
+
+public abstract class WasmSymbolicatorBase
+{
+    public string? SymbolsFile { get; private set; }
+    public string? SymbolsPatternFile { get; private set; }
+
+    public virtual bool Init(string? symbolMapFile, string? symbolsPatternFile, ILogger logger)
+    {
+        SymbolsFile = symbolMapFile;
+        SymbolsPatternFile = symbolsPatternFile;
+
+        return true;
+    }
+
+    public static WasmSymbolicatorBase? Create(Type? symbolicatorType, string? symbolMapFile, string? symbolsPatternFile, ILogger logger)
+    {
+        if (symbolicatorType is null && symbolMapFile is null && symbolsPatternFile is null)
+            return null;
+
+        if (symbolicatorType is null)
+        {
+            logger.LogWarning($"No symbolicator given");
+            return null;
+        }
+
+        var symbolicator = Activator.CreateInstance(symbolicatorType) as WasmSymbolicatorBase;
```

Comment:
Just curious here: the `LoadFromAssemblyPath` will load the assembly into the main context because we put the assembly in the probing path?

Or how come you're able to instantiate `WasmSymbolicatorBase`?

(I am asking about this because I am fighting a similar problem elsewhere and wondering if you know about these things?)

Reply:
We are using the full path to the assembly, so that will get loaded from there. It's assembly dependencies are:
```
        Name=System.Runtime
        Name=Microsoft.Extensions.Logging.Abstractions
        Name=Microsoft.Extensions.Logging.Console
        Name=Microsoft.Extensions.Logging
        Name=Microsoft.DotNet.XHarness.Common
        Name=System.Collections
        Name=System.Text.RegularExpressions
        Name=System.Console
        Name=System.Linq
```

I think these are already loaded in the context. And if there were some that weren't available with xharness, then I would expect them to get probed in the directory containing the symbolicator assembly.